lib/active_remote/typecasting/date_typecaster.rb

```diff
@@ -0,0 +1,10 @@
+module ActiveRemote
+  module Typecasting
+    class DateTypecaster
+      def self.call(value)
+        value.to_date if value.respond_to? :to_date
+      rescue NoMethodError, ArgumentError
```

Comment:
What if we just rescued this instead of checking `respond_to?`? I'm also not sure why `ActiveAttr` only provides this on the date typecaster.